src/odemis/acq/align/tdct.py

```diff
@@ -0,0 +1,146 @@
+"""
+Created on 8 Jan 2025
+
+Copyright © 2025 Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the
+terms  of the GNU General Public License version 2 as published by the Free
+Software  Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY;  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR  PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+
+import logging
+import sys
+from typing import Dict, List, Tuple
+
+import numpy
+import yaml
+
+from odemis import model
+
+# tmp: append the path to the 3DCT module until installation solution
+sys.path.append("/home/patrick/development/openfibsem/3DCT")
+sys.path.append("/home/patrick/development/openfibsem/3DCT/tdct")
+from tdct.correlation_v2 import run_correlation
+from tdct.util import multi_channel_get_z_guass
+
+def convert_das_to_numpy_stack(das: List[model.DataArray]) -> numpy.ndarray:
+    """Convert a list of DataArrays to a numpy stack."""
+    arr = []
+    for da in das:
+        if isinstance(da, model.DataArrayShadow):
+            da = da.getData()
+        if da.ndim == 5: # model.DataArray is always 5D
+            da = da[0, 0, :, :, :]
+        arr.append(da)
+
+    return numpy.stack(arr, axis=0)
+
+def get_optimized_z_guass(das: List[model.DataArray], x: int, y: int, z: int, show: bool = False) -> Tuple[float, float, float]:
+    """Get the best fitting z-coordinate for the given x, y coordinates. Supports multi-channel images.
+    :param das: the data arrays (one per channels)
+    :param x: the x-coordinate
+    :param y: the y-coordinate
+    :param z: the z-coordinate (initial guess)
+    :param show: show the plot for debugging
+    :return: the z-coordinate and the corresponding correlation value"""
```

Comment:
The code looks like it's returning just 1 value. The typing info shows 3 values. This doc says 2 values. Can you fix the discrepancy?